apps/web/views/profile/components/ProfileModal.tsx

```diff
@@ -9,34 +7,31 @@ import {
     DialogHeader,
     DialogTitle,
 } from '@app/ui/components/dialog';
-import { Input } from '@app/ui/components/input';
+import { useAuth } from '@app/hooks';
 import { Label } from '@app/ui/components/label';
-import { useRouter } from 'next/router';
-import React, { useState } from 'react';
+import { Input } from '@app/ui/components/input';
+import { Button } from '@app/ui/components/button';
+import { useProfileUpdate } from '@app/hooks/api/Profile';
 
 type IModal = {
     isOpen: boolean;
     onClose: () => void;
 };
 
 export const ProfileModal = ({ isOpen, onClose }: IModal) => {
-    const { user, setUser } = useAuth();
-    const [name, setName] = useState(user?.displayName);
-    const [slug, setSlug] = useState(user?.slug);
+    const { user } = useAuth();
+    const handleProfileUpdates = useProfileUpdate();
+    // TODO
+    // MOVE TO HOOK FORM ?
+    const [name, setName] = useState<string>(user?.displayName as string);
+    const [slug, setSlug] = useState<string>(user?.slug as string);
```

Comment:
should i use hook form here , i am confued @DarkPhoenix2704  ?

Reply:
ok